src/pages/experiments/[experimentId]/plots-and-tables/marker-heatmap/index.jsx

```diff
@@ -0,0 +1,307 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Row, Col, Space, Collapse, Skeleton, Empty, Typography,
+} from 'antd';
+import _ from 'lodash';
+import { useSelector, useDispatch } from 'react-redux';
+import { Vega } from 'react-vega';
+import PropTypes from 'prop-types';
+import loadProcessingSettings from '../../../../../redux/actions/experimentSettings/processingConfig/loadProcessingSettings';
+import PlotStyling from '../../../../../components/plots/styling/PlotStyling';
+import { updatePlotConfig, loadPlotConfig } from '../../../../../redux/actions/componentConfig';
+import Header from '../../../../../components/plots/Header';
+import { generateSpec } from '../../../../../utils/plotSpecs/generateHeatmapSpec';
+import { loadGeneExpression, loadMarkerGenes, setGeneOrder } from '../../../../../redux/actions/genes';
+import { loadCellSets } from '../../../../../redux/actions/cellSets';
+import PlatformError from '../../../../../components/PlatformError';
+import Loader from '../../../../../components/Loader';
+import populateHeatmapData from '../../../../../components/plots/helpers/populateHeatmapData';
+import HeatmapControls from '../../../../../components/plots/styling/heatmap/HeatmapControls';
+
+const { Text } = Typography;
+const { Panel } = Collapse;
+
+const plotUuid = 'markerHeatmapPlotMain';
+const plotType = 'markerHeatmap';
+
+const route = {
+  path: 'marker-heatmap',
+  breadcrumbName: 'Marker-Heatmap',
+};
+const HeatmapPlot = ({ experimentId }) => {
+  const dispatch = useDispatch();
+  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+  const { expression: expressionData } = useSelector((state) => state.genes);
+  const { error, loading } = expressionData;
+  const cellSets = useSelector((state) => state.cellSets);
+  const { hierarchy, properties } = cellSets;
+  const selectedGenes = useSelector((state) => state.genes.expression.views[plotUuid]?.data) || [];
+  const [currentSearchedGenes, setCurrentSearchedGenes] = useState([]);
+  const {
+    loading: loadingMarkerGenes,
+    error: errorMarkerGenes, order,
+  } = useSelector((state) => state.genes.markers);
+  const [vegaSpec, setVegaSpec] = useState();
+  const louvainClustersResolution = useSelector(
+    (state) => state.experimentSettings.processing
+      .configureEmbedding?.clusteringSettings.methodSettings.louvain.resolution,
+  ) || false;
+
+  useEffect(() => {
+    if (!louvainClustersResolution) {
+      dispatch(loadProcessingSettings(experimentId));
+    }
+    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    dispatch(loadCellSets(experimentId));
+  }, []);
+
+  useEffect(() => {
+    if (louvainClustersResolution && config && !config.selectedGenes.length) {
+      dispatch(loadMarkerGenes(experimentId, louvainClustersResolution, plotUuid, config.numGenes));
+    }
+  }, [louvainClustersResolution, hierarchy, config?.numGenes]);
+
+  const sortGenes = (newGenes) => {
```

Comment:
can you add unit tests to check for whether this sorting works, also to verify that we are not sorting when we are removing a gene and any other case you can come up with?